src/urllib3/poolmanager.py

```diff
@@ -459,9 +461,21 @@ def _set_proxy_headers(self, url, headers=None):
             headers_.update(headers)
         return headers_
 
+    def _validate_proxy_scheme_url_selection(self, url_scheme):
+        if url_scheme == "https" and self.proxy.scheme == "https":
+            warnings.warn(
```

Comment:
Is there a sensible stacklevel to use here?

Reply:
Probably! I'll add one.

Reply:
Settled with 3 so the traceback lands on wherever `urlopen()` is called. There's not really a great way to do [multi-entrypoint warning stacklevel](https://twitter.com/sethmlarson/status/1245074746645057542). :(

Reply:
Yeah, that tweet is how I found out about stacklevel. In my case I'm using `urllib3.ProxyManager` and stacklevel=5 ends up in my code